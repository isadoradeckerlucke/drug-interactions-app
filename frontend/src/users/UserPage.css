div.list-group-item {
    width: 50%;
}

div {
    flex-direction: column;
    align-items: center;
    justify-content: center;
}