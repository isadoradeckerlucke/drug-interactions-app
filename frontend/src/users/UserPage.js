import React from "react";
import { useSelector, useDispatch } from "react-redux";
import { useParams } from "react-router-dom";
import DrugInteractionApi from "../Api";
import { setCurrentUser } from "../actions/currentUser";
import Button from "react-bootstrap/Button";
import ListGroup from "react-bootstrap/ListGroup";
import "./UserPage.css";

function UserPage() {
  const { username } = useParams();
  const dispatch = useDispatch();

  const currentUser = useSelector((st) => st.currentUser);
  // if they're not logged in as this user, show error page
  if (currentUser.isLoggedIn !== true) {
    return (
      <h1>you must be logged in as {username} in order to view this page</h1>
    );
  } else if (currentUser.currentUser.username !== username) {
    return (
      <h1>you must be logged in as {username} in order to view this page</h1>
    );
  }
  const savedDrugs = currentUser.currentUser.savedDrugs;
  async function unsave(e) {
    e.preventDefault();
    await DrugInteractionApi.unsaveDrug(username, e.target.value);
    dispatch(setCurrentUser(username));
  }

  // otherwise show a list of their saved drugs, and each should have a button to unsave
  return (
    <div>
      <h1>{username}'s saved drugs</h1>
      {savedDrugs.length === 0 ? (
        <p>you haven't saved any drugs yet!</p>
      ) : (
        <ListGroup id="list-group">
          {savedDrugs.map((d) => (
            <ListGroup.Item key={d}>
              {d}{" "}
              <Button
                onClick={unsave}
                value={d}
                size="sm"
                variant="outline-danger"
              >
                {"  "}X{"  "}
              </Button>
            </ListGroup.Item>
          ))}
        </ListGroup>
      )}
    </div>
  );
}

export default UserPage;
