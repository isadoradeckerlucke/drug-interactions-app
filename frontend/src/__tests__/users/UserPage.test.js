import React from "react";
import { render } from "@testing-library/react";
import UserPage from "../../users/UserPage";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import { MemoryRouter, Route } from "react-router-dom";

describe("users page works as expected", function () {
  const currentUserTestLoggedIn = {
    isLoggedIn: true,
    currentUser: {
      username: "testuser",
      email: "testuser@gmail.com",
      savedDrugs: [],
    },
  };
  const currentUserTestLoggedOut = {
    isLoggedIn: false,
    currentUser: {},
  };

  const initialStateLoggedIn = {
    drugs: { interactions: {} },
    loading: { loading: false },
    currentUser: currentUserTestLoggedIn,
  };

  const initialStateLoggedOut = {
    drugs: { interactions: {} },
    loading: { loading: false },
    currentUser: currentUserTestLoggedOut,
  };

  const mockStore = configureStore();
  let store;

  /** smoke test */
  it("renders without crashing", function () {
    store = mockStore(initialStateLoggedIn);
    render(
      <MemoryRouter>
        <Provider store={store}>
          <UserPage />
        </Provider>
      </MemoryRouter>
    );
  });

  /** snapshot test */
  it("matches initial snapshot", function () {
    store = mockStore(initialStateLoggedIn);

    const { asFragment } = render(
      <MemoryRouter initialEntries={["/users/test3"]}>
        <Provider store={store}>
          <Route path="/users/:username">
            <UserPage />
          </Route>
        </Provider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
