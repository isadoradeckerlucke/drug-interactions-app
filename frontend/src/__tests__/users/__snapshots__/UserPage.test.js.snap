// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`users page works as expected matches initial snapshot 1`] = `
<DocumentFragment>
  <h1>
    you must be logged in as test3 in order to view this page
  </h1>
</DocumentFragment>
`;
